os/mm/mm_heap/mm_partition_mgr.c

```diff
@@ -0,0 +1,159 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <tinyara/mm/mm.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+/* If multiple regions are supported or if multi-heap is supported in the project,
+then we dont know how to configure the partitions for user applications. Hence we
+will throw an error here
+*/
+
+#if defined(CONFIG_APP_BINARY_SEPARATION) && defined(__KERNEL__)
+
+#if CONFIG_MM_REGIONS > 1
+#error "Unable to handle APP RAM PARTITIONS for CONFIG_MM_REGIONS > 1"
+#endif
+
+#if CONFIG_MM_NHEAPS > 1
+#error "Unable to handle APP RAM PARTITIONS for CONFIG_MM_NHEAPS > 1"
+#endif
+
+/****************************************************************************
+ * Public Variables
+ ****************************************************************************/
+extern uint32_t __ksram_segment_start__[];
+extern uint32_t __ksram_segment_size__[];
+
+/****************************************************************************
+ * Private variables
+ ****************************************************************************/
+static struct mm_partition_s g_mempartitions[CONFIG_NUM_APPS];
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: mm_initialize_partitions
+ *
+ * Description:
+ * This function is called during os boot to initialize the memory partitions.
+ *
+ * Parameters:
+ *   None
+ *
+ * Return Value:
+ *   None
+ *
+ * Assumptions:
+ *
+ ****************************************************************************/
+
+void mm_initialize_partitions(void)
+{
+	int i = 0;
+	uint32_t start;
+	uint32_t size;
+
+	start = (uint32_t)__ksram_segment_start__ + (uint32_t)__ksram_segment_size__;
+	size = (REGION_SIZE - start) / CONFIG_NUM_APPS;
```

Comment:
Should we align the memory address here?

Reply:
There is no requirement for alignment of the partitions. Internally, alignment will be done as required, when heap or other sections are created.

Reply:
Query is when you start the address, it is ok to start with odd address? Since you dividing by number of apps, start address will may no be aligned. Is that ok?

Reply:
Okay. I will make sure to align it to even address.

Reply:
Correction*
size = (REGION_SIZE - (uint32_t)__ksram_segment_size__) / CONFIG_NUM_APPS;
OR
size =  (uint32_t)__usram_segment_size__ / CONFIG_NUM_APPS;

Reply:
ok.. will change it.